conjureup/controllers/deploy/gui.py

```diff
@@ -46,17 +46,14 @@ def init_machines_assignments(self):
         else:
             self.sync_assignments()
 
-    def _handle_exception(self, tag, exc):
-        track_exception(exc.args[0])
-        app.ui.show_exception_message(exc)
-        self.showing_error = True
-        EventLoop.remove_alarms()
-
     def _pre_deploy_exec(self):
         """ runs pre deploy script if exists
         """
-        app.env['JUJU_PROVIDERTYPE'] = model_info(
-            app.current_model)['provider-type']
+        try:
+            app.env['JUJU_PROVIDERTYPE'] = model_info(
+                app.current_model)['provider-type']
+        except LookupError as e:
+            return handle_exception(e)
```

Comment:
why only handle this specific exception? We probably need to stop if any other exception is raised also, right?